### PR TITLE
refactor: trainingNumberEsEventListener to use ProgrammeMembershipRepository.

### PR DESCRIPTION
**Pull request by hihilary**
The initial refacotring from PM to CM is in this commit: https://github.com/Health-Education-England/TIS-TCS/commit/3fd3540

`TrainingNumber` is deprecated in CM, so we have to change the listener to use PM.

However, there's no entry to modify training numbers on TIS admin UI. And, in tcs-service, I'm not able to find anywhere who sends the `TrainingNumberXXXEvent`. Had a quick discussion with James before, and he also mentioned this functionality would be added in the future.

In this PR, I added a query in `ProgrammeMembershipRepository` as I thought we need it, but the current process doesn't modify the `trainingNumber` in PMs, which needs to be fixed when we start to implement this functionality.

TIS21-2388

### CODE REVIEW

File: tcs-persistence/src/test/java/com/transformuk/hee/tis/tcs/service/repository/ProgrammeMembershipRepositoryIntTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package com.transformuk.hee.tis.tcs.service.repository;
```
Comment (jayanta2018) [inline tcs-persistence/src/test/java/com/transformuk/hee/tis/tcs/service/repository/ProgrammeMembershipRepositoryIntTest.java:1]: It is a new file. Just wondering do we need to have the license declaration at the beginning?

File: tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/listener/person/TrainingNumberElasticSearchEventListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
+package com.transformuk.hee.tis.tcs.service.listener.person;
```
Comment (jayanta2018) [inline tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/listener/person/TrainingNumberElasticSearchEventListenerTest.java:1]: May be the same here

File: tcs-persistence/src/test/java/com/transformuk/hee/tis/tcs/service/repository/ProgrammeMembershipRepositoryIntTest.java (hunk 1 above)
Comment (hihilary) [reply tcs-persistence/src/test/java/com/transformuk/hee/tis/tcs/service/repository/ProgrammeMembershipRepositoryIntTest.java:1]: I can't see any license declaration in any of other tcs files, so think it's not needed.

File: tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/listener/person/TrainingNumberElasticSearchEventListenerTest.java (hunk 2 above)
Comment (hihilary) [reply tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/listener/person/TrainingNumberElasticSearchEventListenerTest.java:1]: I can't see any license declaration in any of other tcs files, so think it's not needed.

**Comment by hihilary** [comment]
> I have looked for the rationale for changing from searching `curriculumMembershipRepository.findByTraineeId()` to `programmeMembershipRepository.trainingNumberId()` rather than `programmeMembershipRepository.findByPersonId()` but couldn't see it.
> 
> Based on the data model it looks like a bug but I'm not certain.

TrainingNumber is a separate table in tcs (like some metadata), and every ProgrammeMembership has a nested TrainingNumber object. (CurriculumMembership.TrainingNumber is deprecated)
I think the initial purpose of this listener should be when a training number details is modified, all the nested TrainingNumbers in ProgrammeMembership have to be modified accordingly.

So what I've done here is to get the ProgrammeMemberships which needs a modification. However, currently, we don't modify anything.

**Review by JosephKelly — APPROVED** [review]
Cool.  Squashing probably bugs too :-)

**Pull request merged**